packages/core/package.json

```diff
@@ -47,16 +47,18 @@
   },
   "homepage": "https://github.com/segmentio/analytics-react-native#readme",
   "dependencies": {
+    "@segment/inspector-core": "^2.0.2",
     "@segment/sovran-react-native": "^1",
     "deepmerge": "^4.2.2",
     "js-base64": "^3.7.2",
+    "socket.io-client": "^4.6.2",
```

Comment:
My concern right now is that `socket.io` seems to be a huge library.  1MB+ in size.

Metro doesn't do Tree Shaking so it can be a lot of imports that we don't use (and possibly users won't either in prod builds)

Can we refactor this into a standalone package before publishing it?

An idea is to dispatch events at all the places where we currently do a `inspector.trace` call, the inspector can then subscribe to those events. It could retain the same APIs and `TraceData` object for simplicity.

If we refactor it into a standalone plugin lib we could even dismiss the `debug` check I mentioned in a comment below as we could refer people to only install it as `devDependency`.